common/src/warp_runner/manager/commands/blink_commands.rs

```diff
@@ -147,5 +160,29 @@ pub async fn handle_blink_cmd(cmd: BlinkCmd, blink: &mut Calling) {
                 let _ = rsp.send(blink.disable_automute());
             }
         }
+        BlinkCmd::TestSpeaker { rsp } => {
+            match blink.get_audio_device_config().await {
+                Ok(audio_config) => {
+                    let _ = audio_config
+                        .test_speaker(rsp)
+                        .map_err(warp::error::Error::Any);
+                }
+                Err(e) => {
+                    log::debug!("speaker testing fail {:}", e);
+                }
+            };
+        }
+        BlinkCmd::TestMicrophone { rsp } => {
+            match blink.get_audio_device_config().await {
+                Ok(audio_config) => {
+                    let _ = audio_config
+                        .test_microphone(rsp)
+                        .map_err(warp::error::Error::Any);
+                }
+                Err(e) => {
+                    log::debug!("microphone testing fail {:}", e);
+                }
+            };
```

Comment:
I believe it would be better to use `tokio::task::spawn_blocking(|| {audio_config.test_speaker/test_microphone});` because those functions use `std::thread::sleep()` and will block the tokio runtime.